json_exerciseAnswers.ipynb

```diff
@@ -0,0 +1,619 @@
+      "Social protection and risk management           158\n",
+      "Financial and private sector development        130\n",
+      "Social dev/gender/inclusion                     119\n",
+      "Trade and integration                            72\n",
+      "Urban development                                47\n",
+      "Economic management                              33\n",
+      "Name: name, dtype: int64\n"
+     ]
+    }
+   ],
+   "source": [
+    "#Question 2:\n",
+    "#read in the data as a json\n",
+    "data = json.load((open('data/world_bank_projects.json')))\n",
+    "# use normalization to create tables from nested element\n",
+    "df = json_normalize(data, 'mjtheme_namecode')\n",
+    "\n",
+    "# get rid of any rows that don't have a project theme\n",
+    "result = df[df.name != \"\"]\n",
+    "\n",
+    "# pull out top 10 project themes\n",
+    "results = result.name.value_counts().head(10)\n",
+    "\n",
+    "#print results\n",
+    "print(results)\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Question 3:\n",
+    "#read in the data as a json\n",
+    "data = json.load((open('data/world_bank_projects.json')))\n",
+    "# use normalization to create tables from nested element\n",
+    "df = json_normalize(data, 'mjtheme_namecode')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 94,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Environment and natural resources management    250\n",
+      "Rural development                               216\n",
+      "Human development                               210\n",
+      "Public sector governance                        199\n",
+      "Social protection and risk management           168\n",
+      "Financial and private sector development        146\n",
+      "Social dev/gender/inclusion                     130\n",
+      "Trade and integration                            77\n",
+      "Urban development                                50\n",
+      "Economic management                              38\n",
+      "Name: name, dtype: int64\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "C:\\Users\\BestBuy\\Anaconda2\\lib\\site-packages\\ipykernel\\__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)\n",
+      "  if __name__ == '__main__':\n"
+     ]
+    }
+   ],
+   "source": [
+    "#sort the data based on code number\n",
+    "dfs = df.sort('code')\n",
+    "\n",
+    "#replace the empty spaces with NaN\n",
+    "dfs['name'].replace('', np.nan, inplace=True)\n",
+    "\n",
+    "#use .fillna forward fill to fill in the theme names\n",
+    "fill = dfs.fillna(method = 'ffill')\n",
```

Comment:
Is [this](https://gist.github.com/amirziai/f29686ab7d8d2da70d1020e34609f0f3) helpful?